### PR TITLE
libxml2 recursive files

### PR DESCRIPTION
**Pull request by JaiXJM-BB**

### CODE REVIEW

**Comment by JaiXJM-BB** [comment]
Test results: 
[libxml2_tests.out.txt](https://github.com/user-attachments/files/18042982/libxml2_tests.out.txt)


These are test files located in libxml, but do not seem to have any accompanying instructions. Need more research to see why they are failing. Moreover, xmlconf tests seem to be very outdated (cause of SIGSEGV?)

**Comment by JaiXJM-BB** [comment]
Rebased

**Comment by JaiXJM-BB** [comment]
Meant to force push to trigger workflows but pushed the non-rebased version instead, whoops

File: ports/libxml2/README.md [hunk 1]
```diff
@@ -26,8 +33,130 @@ cd libxml2-2.13.5
...
+```bash
+#QNX 8.0 will be in the directory ~/qnx800/
+#QNX 7.1 will be in the directory ~/qnx710/
+source ~/qnx800/qnxsdp-env.sh
+```
+
+5. Build the project in your workspace from Step 1
+```bash
+# Navigate back to your workspace
+cd <path-to-your-workspace.
```
Comment (eleir9268) [inline ports/libxml2/README.md:85]: typo

File: ports/libxml2/README.md [hunk 2]
```diff
@@ -26,8 +33,130 @@ cd libxml2-2.13.5
...
+TARGET_USER_FOR_INSTALL="qnxuser"
+# Set to the target's IP address
+TARGET_IP_ADDRESS=<target-ip-address>
+
+# Create needed directories on the target
+ssh  $TARGET_USER_FOR_INSTALL@$TARGET_IP_ADDRESS "mkdir -p /data/home/$TARGET_USER_FOR_INSTALL/libxml2/lib"
+ssh  $TARGET_USER_FOR_INSTALL@$TARGET_IP_ADDRESS "mkdir -p /data/home/$TARGET_USER_FOR_INSTALL/libxml2/test/.lib"
+
+# Change aarch64 to x86_64 depending on your architecture.
+scp -r $QNX_TARGET/aarch64le/usr/lib/libxml* $TARGET_USER_FOR_INSTALL@$TARGET_IP_ADDRESS:/data/home/$TARGET_USER_FOR_INSTALL/libxml2/lib
```
Comment (eleir9268) [inline ports/libxml2/README.md:140]: `-r` isn't needed

Comment (JaiXJM-BB) [reply ports/libxml2/README.md]: Fixed with 3f3fbcad47bbc8b0175bd94bf19e3f3aeea1ff4d!

File: ports/libxml2/README.md (hunk 1 above)
Comment (JaiXJM-BB) [reply ports/libxml2/README.md]: Fixed with 3f3fbcad47bbc8b0175bd94bf19e3f3aeea1ff4d 👍

**Review by eleir9268 — APPROVED** [review]

**Comment by JaiXJM-BB** [comment]
Updated to point to the GNOME repo instead of our own fork, since changes are upstreamed. 
This WILL break the build_source test since it needs a higher automake version.
Relevant Commits:
f33f70de3ee8f6df06fb873ab88fdd2e9388d47f
9a7f9567e9d087dbcef99a70cf7bc740cdf614a2

**Comment by chachoi** [comment]
Can we simply make the Dockerfile we have install a higher version of automake to make the source build work?

**Comment by JaiXJM-BB** [comment]
not sure why trying to edit the dockerfile to update autoconf broke the README, but that's why I tred to undo it via force push

**Comment by JaiXJM-BB** [comment]
Fixed :)

**Comment by JaiXJM-BB** [comment]
Looks like the source build check worked. I just used sed to edit the minimum required automake version (not a great solution but it'll have to do until we update the docker ubuntu version)

**Comment by JaiXJM-BB** [comment]
re-requesting review due to changes regarding upstreaming (did not think it would happen prior to this PR being merged)

**Comment by eleir9268** [comment]
I think we should just have a second docker image. I don't think we can realistically expect to be able to build everything with just one image in the long run.

**Comment by JaiXJM-BB** [comment]
> I think we should just have a second docker image. I don't think we can realistically expect to be able to build everything with just one image in the long run.

I agree, but I think we need a bigger discussion on how many tests we want to run, whether we should include actual unit tests & on what platforms (Ubuntu versions etc), concurrency management, and general scope of our workflows once the new testing machines come in. We should make changes regarding that all at once in a future PR so it's easier to keep track of.

**Comment by eleir9268** [comment]
> > I think we should just have a second docker image. I don't think we can realistically expect to be able to build everything with just one image in the long run.
> 
> I agree, but I think we need a bigger discussion on how many tests we want to run, whether we should include actual unit tests & on what platforms (Ubuntu versions etc), concurrency management, and general scope of our workflows once the new testing machines come in. We should make changes regarding that all at once in a future PR so it's easier to keep track of.

Ok, as long as that's on the radar. Currently people will encounter the automake error whenever they try to follow the instructions for building from source in a docker container.

**Comment by JaiXJM-BB** [comment]
> > I agree, but I think we need a bigger discussion on how many tests we want to run, whether we should include actual unit tests & on what platforms (Ubuntu versions etc), concurrency management, and general scope of our workflows once the new testing machines come in. We should make changes regarding that all at once in a future PR so it's easier to keep track of.
> 
> Ok, as long as that's on the radar. Currently people will encounter the automake error whenever they try to follow the instructions for building from source in a docker container.

Great point. For now I've made a note in the README and a line on how to fix it, and combined the docker/ubuntu instructions into one set with the docker step listed as optional so that it is more readable and clear. afba9aa0dc9451243e547d72691e86bbc73b3cf4

**Review by eleir9268 — APPROVED** [review]

**Pull request merged**